Build: qt6 build issues

With Qt 6.2, MSVC 2019, and Qt Creator 10.01 on Windows 10 64-bit (Desktop_Qt_6_2_4_MSVC2019_64bit)
The following issues appear during compiling:

- seamly2d\src\libs\vmisc\vabstractapplication.cpp:283: warning: C4834: discarding return value of function with 'nodiscard' attribute
- seamly2d\src\libs\vmisc\vabstractapplication.cpp:284: warning: C4834: discarding return value of function with 'nodiscard' attribute
- seamly2d\src\libs\vmisc\vabstractapplication.cpp:285: warning: C4834: discarding return value of function with 'nodiscard' attribute
- seamly2d\src\libs\vmisc\vabstractapplication.cpp:292: warning: C4834: discarding return value of function with 'nodiscard' attribute
- seamly2d\src\libs\vmisc\vabstractapplication.cpp:293: warning: C4834: discarding return value of function with 'nodiscard' attribute
- seamly2d\src\app\seamly2d\dialogs\export_layout_dialog.cpp:766: warning: C4003: not enough arguments for function-like macro invocation 'QStringLiteral'
- seamly2d\src\app\seamly2d\dialogs\export_layout_dialog.cpp:766: warning: C4003: not enough arguments for function-like macro invocation 'QT_UNICODE_LITERAL'
- seamly2d\src\app\seamly2d\dialogs\export_layout_dialog.cpp:88: warning: Definition of implicit copy constructor for 'QException' is deprecated because it has a user-provided destructor
- seamly2d\src\app\seamly2d\dialogs\export_layout_dialog.cpp:88: In implicit copy constructor for 'QException' first required here
- seamly2d\src\app\seamly2d\dialogs\export_layout_dialog.cpp:782: warning: Default label in switch which covers all enumeration values
- seamly2d\src\app\seamly2d\dialogs\export_layout_dialog.cpp:34: warning: Included header options.h is not used directly (fix available)

I took a closer look at all these:

- seamly2d\src\libs\vmisc\vabstractapplication.cpp:283: warning: C4834: discarding return value of function with 'nodiscard' attribute
- seamly2d\src\libs\vmisc\vabstractapplication.cpp:284: warning: C4834: discarding return value of function with 'nodiscard' attribute
- seamly2d\src\libs\vmisc\vabstractapplication.cpp:285: warning: C4834: discarding return value of function with 'nodiscard' attribute
- seamly2d\src\libs\vmisc\vabstractapplication.cpp:292: warning: C4834: discarding return value of function with 'nodiscard' attribute
- seamly2d\src\libs\vmisc\vabstractapplication.cpp:293: warning: C4834: discarding return value of function with 'nodiscard' attribute

![nodiscard](https://github.com/FashionFreedom/Seamly2D/assets/31944718/6348b3d5-1fb3-4176-a835-3c88b8fc9cc5)

 I would assume  it's complaing about the load() functions which return a bool that are not being checked. 

These are legit warnings as it should be calling QStringLiteral("") instead of QStringLiteral()

- seamly2d\src\app\seamly2d\dialogs\export_layout_dialog.cpp:766: warning: C4003: not enough arguments for function-like macro invocation 'QStringLiteral'
- seamly2d\src\app\seamly2d\dialogs\export_layout_dialog.cpp:766: warning: C4003: not enough arguments for function-like macro invocation 'QT_UNICODE_LITERAL'

This is also legit... since the base class already has a virtual destructor, no need to redefine one in the subclass - thus it's "deprecated" 

- seamly2d\src\app\seamly2d\dialogs\export_layout_dialog.cpp:88: warning: Definition of implicit copy constructor for 'QException' is deprecated because it has a user-provided destructor

This one is a bit confusing as like I stated before  there are mulitple instances of the use of a case followed by default:. This appears to be complaining because it expects the previous case to break, then the default case. Simple solution in this case is to simply remove the default label. Shouldn't be calling the routine with a case that can't be handled anyway... and in the event some other programmer does, they'll get an error meaning they need to fix something. 

- seamly2d\src\app\seamly2d\dialogs\export_layout_dialog.cpp:782: warning: Default label in switch which covers all enumeration values

Again this is one is just pointing out the include can be removed as it's not needed. Which I'm sure theres a lot more than just this unused include. 

- seamly2d\src\app\seamly2d\dialogs\export_layout_dialog.cpp:34: warning: Included header options.h is not used directly (fix available)

I will fix all these.

--- a/src/app/seamly2d/dialogs/export_layout_dialog.cpp
+++ b/src/app/seamly2d/dialogs/export_layout_dialog.cpp
@@ -821,7 +821,7 @@ void ExportLayoutDialog::writeSettings() const
  */
 QString ExportLayoutDialog::modeString() const
 {
-    QString modeStr = QStringLiteral();
+    QString modeStr = QStringLiteral("");
     if (qApp->Seamly2DSettings()->useModeType())
     {
         switch (m_mode)

--- a/src/libs/tools/images/image_tool.cpp
+++ b/src/libs/tools/images/image_tool.cpp
@@ -63,11 +63,11 @@ QString getImageFilename(QWidget *parent)
 
     const QString path = qApp->Seamly2DSettings()->getImageFilePath();
 
-    bool usedNotExistedDir = false;
     QDir directory(path);
     if (!directory.exists())
     {
-        usedNotExistedDir = directory.mkpath(".");
+        // Explicitly cast to void to suppress clang warnings
+        (void)directory.mkpath(".");
     }
 
     const QString filename = QFileDialog::getOpenFileName(parent, QObject::tr("Open Image File"), path, filter, nullptr,

--- a/src/libs/vmisc/vabstractapplication.cpp
+++ b/src/libs/vmisc/vabstractapplication.cpp
@@ -279,17 +279,20 @@ void VAbstractApplication::loadTranslations(const QString &locale)
     pmsTranslator    = new QTranslator(this);
 
 #if defined(Q_OS_WIN) || defined(Q_OS_MAC)
-    qtTranslator->load("qt_" + locale, translationsPath(locale));
-    qtxmlTranslator->load("qtxmlpatterns_" + locale, translationsPath(locale));
-    qtBaseTranslator->load("qtbase_" + locale, translationsPath(locale));
+    // Explicitly cast to void to suppress clang warnings
+    (void)qtTranslator->load("qt_" + locale, translationsPath(locale));
+    (void)qtxmlTranslator->load("qtxmlpatterns_" + locale, translationsPath(locale));
+    (void)qtBaseTranslator->load("qtbase_" + locale, translationsPath(locale));
 #else
-    qtTranslator->load("qt_" + locale, QLibraryInfo::location(QLibraryInfo::TranslationsPath));
-    qtxmlTranslator->load("qtxmlpatterns_" + locale, QLibraryInfo::location(QLibraryInfo::TranslationsPath));
-    qtBaseTranslator->load("qtbase_" + locale, QLibraryInfo::location(QLibraryInfo::TranslationsPath));
+    // Explicitly cast to void to suppress clang warnings
+    (void)qtTranslator->load("qt_" + locale, QLibraryInfo::location(QLibraryInfo::TranslationsPath));
+    (void)qtxmlTranslator->load("qtxmlpatterns_" + locale, QLibraryInfo::location(QLibraryInfo::TranslationsPath));
+    (void)qtBaseTranslator->load("qtbase_" + locale, QLibraryInfo::location(QLibraryInfo::TranslationsPath));
 #endif
 
-    appTranslator->load("seamly2d_" + locale, translationsPath(locale));
-    pmsTranslator->load("measurements_" + locale, translationsPath(locale));
+    // Explicitly cast to void to suppress clang warnings
+    (void)appTranslator->load("seamly2d_" + locale, translationsPath(locale));
+    (void)pmsTranslator->load("measurements_" + locale, translationsPath(locale));
 
     installTranslator(qtTranslator);
     installTranslator(qtxmlTranslator);

--- a/src/libs/vtools/dialogs/support/edit_formula_dialog.cpp
+++ b/src/libs/vtools/dialogs/support/edit_formula_dialog.cpp
@@ -549,10 +549,12 @@ void EditFormulaDialog::SetFormula(const QString &value)
     if (qApp->Settings()->autoClearFx())
     {
         bool isInt;
-        int intValue = m_formula.toInt(&isInt);
+        // Explicitly cast to void to suppress clang warnings
+        (void)m_formula.toInt(&isInt);
 
         bool isDouble;
-        double doubleValue = m_formula.toDouble(&isDouble);
+        // Explicitly cast to void to suppress clang warnings
+        (void)m_formula.toDouble(&isDouble);
 
         if(isInt || isDouble)
         {

